packages/renderer/src/lib/kube/pods/KubernetesTerminalBrowser.spec.ts

```diff
@@ -0,0 +1,154 @@
+/**********************************************************************
+ * Copyright (C) 2024-2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { render, screen } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { beforeAll, beforeEach, expect, test, vi } from 'vitest';
+
+import { podsInfos } from '/@/stores/pods';
+
+import { terminalService } from '../../pod/KubernetesTerminalService';
+import KubernetesTerminalBrowser from './KubernetesTerminalBrowser.svelte';
+import type { PodUI } from './PodUI';
+
+vi.mock('@xterm/xterm', () => {
+  return {
+    Terminal: vi
+      .fn()
+      .mockReturnValue({ loadAddon: vi.fn(), open: vi.fn(), write: vi.fn(), dispose: vi.fn(), onData: vi.fn() }),
+  };
+});
+
+vi.mock('@xterm/addon-serialize');
+
+// dummy event emitter
+const eventEmitter = {
+  receive: (): void => {},
+};
+
+beforeAll(() => {
+  Object.defineProperty(global, 'window', {
+    value: {
+      events: {
+        receive: eventEmitter.receive,
+      },
+      navigator: {
+        clipboard: {
+          writeText: vi.fn(),
+        },
+      },
+      getConfigurationValue: vi.fn(),
+      kubernetesExec: vi.fn(),
+      kubernetesExecResize: vi.fn(),
+      getComputedStyle: vi.fn(),
+      addEventListener: vi.fn(),
+      removeEventListener: vi.fn(),
+      matchMedia: vi.fn().mockReturnValue({
+        matches: false,
+        addEventListener: vi.fn(),
+        removeEventListener: vi.fn(),
+      }),
+    },
+  });
+  vi.mocked(window.getComputedStyle).mockReturnValue({
+    getPropertyValue: vi.fn().mockReturnValue(''),
+  } as unknown as CSSStyleDeclaration);
+});
+
+const podName = 'My Pod';
+
+// create a pod with 2 containers
+const pod: PodUI = {
+  name: podName,
+  selected: true,
+  containers: [
+    {
+      Id: '1234',
+      Names: 'container1',
+      Status: 'running',
+    },
+    {
+      Id: '5678',
+      Names: 'container2',
+      Status: 'running',
+    },
+  ],
+} as unknown as PodUI;
+
+beforeEach(() => {
+  vi.clearAllMocks();
```

Comment:
`clearAllMocks` here, `resetAllMocks` bellow in other component, why?

Reply:
Copied from the equivalent pods test - removed this.